### PR TITLE
Merge the PR on target branch before starting the test

### PR DESCRIPTION
**Pull request by maelk**
This PR adds the logic to merge the tested PR onto the target branch before running the tests. It removes the need for rebasing, and fails if there is a conflict.

### CODE REVIEW

File: jenkins/scripts/files/run_integration_tests.sh [hunk 1]
```diff
@@ -6,10 +6,28 @@ REPO_ORG="${1:-metal3-io}"
...
+# Since we take care of the repo tested here (to merge the PR), do not update
+# the repo in metal3-dev-env 03_launch_mgmt_cluster.sh
+export FORCE_REPO_UPDATE=false
+
+# Clone the source repository
+cd "/home/${USER}"
+git clone ${UPDATED_REPO} tested_repo
+cd tested_repo
+git fetch origin
+git remote add upstream "https://github.com/${REPO_ORG}/${REPO_NAME}.git"
```
Comment (fmuyassarov) [inline jenkins/scripts/files/run_integration_tests.sh]: Isn't it the upstream is going to be the same as the origin (UPDATED_REPO) in this case?

Comment (maelk) [reply jenkins/scripts/files/run_integration_tests.sh]: no, UPDATED_REPO points to the source repo of the PR, for example Nordix, or whatever fork. While the upstream points to the target repo : for example metal3-io metal3-dev-env

**Comment by maelk** [comment]
/hold cancel
Required PRs were merged

**Pull request merged**